src/GraphQL.MemoryCache/MemoryDocumentCacheOptions.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using Microsoft.Extensions.Options;
+
+namespace GraphQL.Caching
+{
+    /// <summary>
+    /// Provides configuration options for <see cref="MemoryDocumentCache"/>.
+    /// </summary>
+    public class MemoryDocumentCacheOptions : IOptions<MemoryDocumentCacheOptions>
+    {
+        /// <summary>
+        /// The maximum total length of all queries cached. Assume maximum memory used is about 10x this value.
+        /// </summary>
+        public int MaxTotalQueryLength { get; set; }
```

Comment:
```suggestion
        public long? MaxTotalQueryLength { get; set; }
```

By the way, `MemoryCacheOptions.SizeLimit` is nullable long.

Reply:
I know

Reply:
Setting the value to null will crash the code when attempting to add an entry with a `Size` value.  Not really sure what value there is to allowing a cache over 20GB.

Reply:
Then why to limit it to not nullable int ?
`MemoryCacheOptions.SlidingExpiration` is also nullable which means that all checks in the ctor are not needed.

Reply:
What if someone wants to have a cache without size limit and no expiration?

Reply:
Making `SlidingExpiration` nullable makes sense.

Reply:
We can set the both to nullable, but the default options would then result in a cache without limit whose entries are never removed from the cache.

Reply:
I think we should force people to set a size limit.  Do you not agree?  They can always derive from the class for unique situations, utilizing the protected members.

Reply:
> Setting the value to null will crash the code when attempting to add an entry with a Size value.

?

Reply:
> I think we should force people to set a size limit. Do you not agree? They can always derive from the class for unique situations, utilizing the protected members.

I do not agree. We **should not** force to derive from our base class just in case of allowing some option. We **should** provide good defaults.

Reply:
I could be wrong...

Reply:
Ok

Reply:
> What if someone wants to have a cache without size limit and no expiration?

Presumably they will crash their computer.

Reply:
Well, so be it.